HKMP/Game/Client/Entity/MantisLord.cs

```diff
@@ -0,0 +1,907 @@
+            }
+
+            if (animation == Animation.ThrowAntic) {
+                _mantisFsm.ExecuteActions("Throw Antic", 1);
+            }
+
+            if (animation == Animation.Throw) {
+                _mantisFsm.FsmVariables.GetFsmBool("Throw R").Value = animationInfo[0] == 1;
+                _mantisFsm.FsmVariables.GetFsmBool("No Dust").Value = animationInfo[1] == 1;
+                _mantisFsm.FsmVariables.GetFsmBool("Throw UD").Value = animationInfo[2] == 1;
+
+                _mantisFsm.ExecuteActions("Throw", 1, 2, 3, 4, 5, 6, 8);
+            }
+
+            if (animation == Animation.Arrive2) {
+                _mantisFsm.ExecuteActions("Arrive 2", 1, 2, 3, 4);
+            }
+
+            if (animation == Animation.ThrowAntic2) {
+                // This one is actually the same as "Throw Antic", but separate here for clarity and host switches
+                _mantisFsm.ExecuteActions("Throw Antic 2", 1);
+            }
+
+            if (animation == Animation.Throw2) {
+                // Check to throw the projectile
+                var shotEvent = (ShotEvent) animationInfo[0];
+                var shotEventFsm = "";
+
+                switch (shotEvent) {
+                    case ShotEvent.WIDE_L:
+                        shotEventFsm = "WIDE L";
+                        break;
+                    case ShotEvent.WIDE_R:
+                        shotEventFsm = "WIDE R";
+                        break;
+                    case ShotEvent.NARROW_L:
+                        shotEventFsm = "NARROW L";
+                        break;
+                    case ShotEvent.NARROW_R:
+                        shotEventFsm = "NARROW R";
+                        break;
+                }
+                _mantisFsm.FsmVariables.GetFsmString("Shot Event").Value = shotEventFsm;
+                _mantisFsm.ExecuteActions("Throw 2", 1, 2, 3, 4);
+            }
+
+            if (animation == Animation.WallLeave) {
+                _mantisFsm.ExecuteActions("Wall Leave 1", 2);
+                _mantisFsm.GetAction<Tk2dPlayAnimationWithEvents>("Wall Leave 1", 1).Execute(() => {
+                    _mantisFsm.GetAction<Tk2dPlayAnimationWithEvents>("Wall Leave 2", 0).Execute(() => {
+                        _mantisFsm.ExecuteActions("After Throw Pause", 1, 2);
+                    });
+                });
+            }
+        }
+
+        public override void UpdateState(byte state) {
+        }
+
+        protected enum State {
+            Idle = 0,
+            Active
+        }
+
+        protected enum Animation {
+            Init = 0,
+            DashArrive,
+            DashAntic,
+            DashAttack,
+            DashRecover,
+            DashLeave,
+            DstabArrive,
+            DstabAttack,
+            DstabLand,
+            DstabLeave,
+            Away,
+            Arrive,
+            ThrowAntic,
+            Throw,
+            HighThrow,
```

Comment:
This enum value is unused, make sure to check whether you didn't miss anything with the implementation.
```suggestion
```

Reply:
Ah I must have left this when I didn't understand how the second phase worked, it has nothing to do with animations so it is not needed.